src/module.c

```diff
@@ -2092,10 +2092,11 @@ void RM_Yield(RedisModuleCtx *ctx, int flags, const char *busy_reply) {
             server.busy_module_yield_reply = busy_reply;
             /* start the blocking operation if not already started. */
             if (!server.busy_module_yield_flags) {
-                server.busy_module_yield_flags = flags & REDISMODULE_YIELD_FLAG_CLIENTS ?
-                    BUSY_MODULE_YIELD_CLIENTS : BUSY_MODULE_YIELD_EVENTS;
+                server.busy_module_yield_flags = BUSY_MODULE_YIELD_EVENTS;
                 blockingOperationStarts();
             }
+            if (flags & REDISMODULE_YIELD_FLAG_CLIENTS)
+                server.busy_module_yield_flags |= BUSY_MODULE_YIELD_CLIENTS;
 
             /* Let redis process events */
             processEventsWhileBlocked();
```

Comment:
I still hope we can check if `RM_Yield` is nested by itself, like adding a  global variable `server.module_yield`:
```c
void RM_Yield(RedisModuleCtx *ctx, int flags, const char *busy_reply) {
    long long now = getMonotonicUs();
    if (now >= ctx->next_yield_time) {
        serverAssert(!server.module_yield);
        server.module_yield = 1;
        ...
            processEventsWhileBlocked();
        server.module_yield = 0;
        ...
    }
}
```

Reply:
we have the `ProcessingEventsWhileBlocked` global we could have used, but it's private to netowrking.c
i guess we can just declare a static variable next to RM_Yield.
WDYT?

Reply:
agree : )